libs/component-library/src/lib/radio-button/RadioButton.tsx

```diff
@@ -0,0 +1,228 @@
+   */
+  uncheckedDisabledColor?: string;
+  /**
+   * Custom color for unchecked disabled radio.
+   */
+  uncheckedDisabledBackgroundColor?: string;
+};
+
+const BORDER_WIDTH = 1;
+const BORDER_WIDTH_CHECKED = 6;
+
+/**
+ * ## Usage
+ * ```js
+ * import * as React from 'react';
+ * import { View } from 'react-native';
+ * import { RadioButton } from '@functionland/component-library';
+ *
+ * const MyComponent = () => {
+ *   const [checked, setChecked] = React.useState('first');
+ *
+ *   return (
+ *     <View>
+ *       <RadioButton
+ *         value="first"
+ *         status={ checked === 'first' ? 'checked' : 'unchecked' }
+ *         onPress={() => setChecked('first')}
+ *       />
+ *       <RadioButton
+ *         value="second"
+ *         status={ checked === 'second' ? 'checked' : 'unchecked' }
+ *         onPress={() => setChecked('second')}
+ *       />
+ *     </View>
+ *   );
+ * };
+ *
+ * export default MyComponent;
+ * ```
+ */
+
+const RadioButton = ({ disabled, onPress, value, status, ...rest }: Props) => {
+  const { colors } = useTheme<FxTheme>();
+  const { value: contextValue, onValueChange } = useRadioButtonContext();
+  const checked =
+    isChecked({
+      contextValue,
+      status,
+      value,
+    }) === 'checked';
+  const { current: borderAnim } = React.useRef<Animated.Value>(
+    new Animated.Value(checked ? BORDER_WIDTH_CHECKED : BORDER_WIDTH)
+  );
+
+  const { current: radioAnim } = React.useRef<Animated.Value>(
+    new Animated.Value(1)
+  );
+
+  const isFirstRendering = React.useRef<boolean>(true);
+
+  React.useEffect(() => {
+    // Do not run animation on very first rendering
+    if (isFirstRendering.current) {
+      isFirstRendering.current = false;
+      return;
+    }
+
+    if (checked) {
+      radioAnim.setValue(2.5);
+
+      Animated.timing(radioAnim, {
+        toValue: 1,
+        duration: 250,
+        useNativeDriver: true,
+      }).start();
+
+      Animated.timing(borderAnim, {
+        toValue: BORDER_WIDTH_CHECKED,
+        duration: 250,
+        useNativeDriver: false,
+      }).start();
+    } else {
+      Animated.timing(borderAnim, {
+        toValue: BORDER_WIDTH,
+        duration: 250,
+        useNativeDriver: false,
+      }).start();
+    }
+  }, [checked, borderAnim, radioAnim]);
+
+  const radioBackgroundColor = disabled
+    ? rest.uncheckedDisabledBackgroundColor || colors.backgroundSecondary
+    : undefined;
+  const checkedColor = rest.checkedColor || colors.greenBase;
+  const uncheckedColor = rest.uncheckedColor || colors.border;
+  const checkedDisabledColor = rest.checkedDisabledColor || colors.greenBorder;
+  const uncheckedDisabledColor = rest.uncheckedDisabledColor || colors.border;
+
+  let radioColor: string;
+  if (disabled) {
+    radioColor = checked ? checkedDisabledColor : uncheckedDisabledColor;
+  } else {
+    radioColor = checked ? checkedColor : uncheckedColor;
+  }
+
+  return (
+    <TouchableBox
+      style={s.container}
```

Comment:
this can just be `borderRadius={}` here since it's a Restyle Box

Reply:
Have updated to use `borderRadius={'l'}`. 

Initially I had build the component without restyle as this was the only instance of it but then decided to give the user restyle functions and added it in.